### PR TITLE
KSTREAMS-6497: Refine CurrentAssignmentBuilderTest

### PR DESCRIPTION
**Pull request by mjsax**
This PR is still WIP, but I have a few question (cf inline comments) and thus opened it to get clarification.

### CODE REVIEW

**Comment by lucasbru** [comment]
@mjsax I couldn't find any inline comments with questions, am I missing something?

File: group-coordinator/src/test/java/org/apache/kafka/coordinator/group/streams/CurrentAssignmentBuilderTest.java [hunk 1]
```diff
@@ -31,177 +33,217 @@
 
 public class CurrentAssignmentBuilderTest {
 
+    // TODO refine test name after semantics of `withCurrentActiveTaskEpoch` are clarified
```
Comment (mjsax) [inline group-coordinator/src/test/java/org/apache/kafka/coordinator/group/streams/CurrentAssignmentBuilderTest.java:36]: @lucasbru I don't understand what the semantics of `withCurrentActiveTaskEpoch` is? Can you explain?

And thus, I don't grog what the tests are doing -- sometime it's set to the "original epoch" but sometimes to `-1` (not sure what `-1` means?).

Comment (lucasbru) [reply group-coordinator/src/test/java/org/apache/kafka/coordinator/group/streams/CurrentAssignmentBuilderTest.java:36]: I checked, I think `withCurrentActiveTaskEpoch` is basically copied from the KIP-848 logic, and initializing the epoch as `-1` means that the previous owner has revoked the active task, so we can assign it to a new one. So `-1` means "currently no owner".

Note that CurrentAssignmentBuilder is not part of KSTREAMS-6497, we have a separate ticket KSTREAMS-6490 which is more involved. We have to update it first to implement the reconciliation also of standby and warm-up tasks as described in the KIP. So, if you decide to work on these tests now, it'll be somewhat incomplete.

**Comment by lucasbru** [comment]
@mjsax any plans to revive this?

**Pull request closed without merging**